components/kyma-environment-broker/internal/broker/instance_update.go

```diff
@@ -348,3 +358,31 @@ func (b *UpdateEndpoint) isKyma2(instance *internal.Instance) (bool, string, err
 	kv := s.GetKymaVersion()
 	return internal.DetermineMajorVersion(kv) == 2, kv, nil
 }
+
+// processExpirationParam returns true, if expired
+func (b *UpdateEndpoint) processExpirationParam(instance *internal.Instance, details domain.UpdateDetails) (*internal.Instance, error) {
+	// if the instance was expired before (in the past), then no need to do any work
+	if instance.IsExpired() {
+		return instance, nil
+	}
+	if len(details.RawParameters) != 0 {
+		var params internal.UpdatingParametersDTO
+		err := json.Unmarshal(details.RawParameters, &params)
+		if err != nil {
+			return instance, err
+		}
+		if params.Expired {
+			if !IsTrialPlan(instance.ServicePlanID) {
+				b.log.Warn("Expiration for non-trial instance is not supported")
+				return instance, nil
```

Comment:
should KEB return an error and information to the caller as well so the consumer realizes expiring non-trial is not supported?

Reply:
yes, you are right

Reply:
fixed